### PR TITLE
Chore:(Docs) Complements the documentation with MDX

### LINKED ISSUE #6725: Addon-docs: When using storybook for angular, cant load docs
**Describe the bug**
I tried to use storybook addon-docs for my angular storybook project. After creating my first mdx story, it fail with the message `SyntaxError: file: Unexpected token (15:9)`, pointing to JSX-syntax: `return <div {...props}/>`.
Thats why I installed `@babel/preset-react`. Now everything loads fine, my template also gets displayed, but unfortunately it fails when I click on Docs with the following StackTrace
```
index.js:47 TypeError: Cannot read property 'toLowerCase' of undefined
    at sanitize (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:87301)
    at sanitizeSafe (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:87308)
    at toId (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:87318)
    at VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:81811
    at updateContextConsumer (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:171400)
    at beginWork (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:171588)
    at performUnitOfWork (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:175228)
    at workLoop (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:175268)
    at renderRoot (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:175351)
    at performWorkOnRoot (VM654 vendors~main.d14197376ef6ce2839d3.bundle.js:176258)
```

**To Reproduce**
1. Create storybook
2. Install addon-docs by using the manual configuration described here: https://www.npmjs.com/package/@storybook/addon-docs
3. Create mdx story with <Story> tag
```
import { Story } from '@storybook/components';

export const componentMeta = {
  title: 'MDX|Button',
};

# Hello Docs

Welcome to the future of Storybook!

<Story name="with text">
  {{
    template: '<div class="button">Button</div>',
  }}
</Story>
```

**Expected behavior**
It should render the given template and the documentation

**Code snippets**
If applicable, add code samples to help explain your problem.

**System:**
 - OS: MacOS
 - Device: Macbook Pro 2018
 - Browser: Chrome
 - Framework: angular
 - Addons: docs
 - Version: 5.2.0-alpha.3

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request, the documentation is expanded to include examples for MDX. 

Addresses and closes #6725 and #15648.

What was done:
- Created snippets for React, Vue, Angular, Svelte.
- Adjusted the docs accordingly.
- Moved the placement of snippets to avoid confusion.
- Fixed a couple of items with current snippets

There's a small caveat to this pull request, as the args composition snippets were not adjusted and is something that can be followed up with another pull request. @shilman I'll follow up on this matter so that we can come up with a good approach and we'll go from there.

Feel free to provide feedback

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@winkerVSbecks care to take a look at this pull request and let me know if you spot anything that needs fixing?

**Comment by winkerVSbecks** [comment]
LGTM

File: docs/snippets/angular/badge-story.mdx.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# Badge
+
+Let's define a story for our `Badge` component:
+
+<Story name="positive" args={{
+  status: 'positive',
+  label: 'Positive'
+}}>
+{Template.bind({})}
```
Comment (shilman) [inline docs/snippets/angular/badge-story.mdx.mdx:20]:
```suggestion
  {Template.bind({})}
```

File: docs/snippets/angular/button-story-component-decorator.mdx.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+<!-- With component which contains ng-content -->
+<Meta
+  title="Components/Button"
+  component={Button}
+  decorators={[
+    moduleMetadata({
+      declarations: [ParentComponent],
+    }),
+     componentWrapperDecorator(ParentComponent),
```
Comment (shilman) [inline docs/snippets/angular/button-story-component-decorator.mdx.mdx:29]:
```suggestion
    componentWrapperDecorator(ParentComponent),
```

File: docs/snippets/angular/checkbox-story.mdx.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      label: 'Unchecked', 
+      checked: true 
+    }}>
+    {Template.bind({})}
+   </Story>
+  <Story name="Secondary" args={{
+    label: 'Secondary', 
+    checked: true, 
+    appearance: 'secondary'
+  }}>
```
Comment (shilman) [inline docs/snippets/angular/checkbox-story.mdx.mdx:33]:
```suggestion
    }}>
```

**Review by shilman — APPROVED** [review]
Looks good! Lots of inconsistent indentation here -- I'm sure I only caught a fraction.

**Comment by jonniebigodes** [comment]
With the latest commits, I've polished and updated the snippets based on feedback and ran through the other mdx examples currently documented and adjusted them to include missing imports, and apply some consistency to them

File: docs/snippets/angular/badge-story.mdx.mdx (hunk 1 above)
Comment (jonniebigodes) [reply docs/snippets/angular/badge-story.mdx.mdx]: Fixed with the latest commits

File: docs/snippets/angular/checkbox-story.mdx.mdx (hunk 3 above)
Comment (jonniebigodes) [reply docs/snippets/angular/checkbox-story.mdx.mdx]: Fixed with the latest commits

File: docs/snippets/angular/button-story-component-decorator.mdx.mdx (hunk 2 above)
Comment (jonniebigodes) [reply docs/snippets/angular/button-story-component-decorator.mdx.mdx]: Fixed with the latest commits

**Review by shilman — COMMENTED** [review]
I think our prettier setup uses single quotes for code but double quotes for JSX attributes. I don't particularly care one way or the other, but we should probably be consistent. Even better, is there some way to get prettier to run on the MDX snippets inside the MDX files?

**Comment by jonniebigodes** [comment]
I'll adjust accordingly! I saw that some of the mdx snippets were using single quotes, for example [here](https://storybook.js.org/docs/react/essentials/actions#gatsby-focus-wrapper) and some other places so I tried a unified approach to this. But I can adjust accordingly based on what you've mentioned. I'll go through them and push a commit in a bit. Regarding your question:
>Even better, is there some way to get prettier to run on the MDX snippets inside the MDX files?

I've had mixed results where it formatted the code properly, other times it broke the example/snippet formatting altogether.

Probably some rule or some misconfiguration that is missing or not being applied properly.

**Comment by jonniebigodes** [comment]
Just ran through the entire mdx snippets and applied the same formatting for all of them.

**Pull request merged**